docs/sources/operators-guide/configuring/reference-configuration-parameters/index.md

```diff
@@ -3397,6 +3397,13 @@ bucket_store:
   # CLI flag: -blocks-storage.bucket-store.posting-offsets-in-mem-sampling
   [postings_offsets_in_mem_sampling: <int> | default = 32]
 
+  index_header:
+    # (experimental) If enabled, the store-gateway will attempt to pre-populate
```

Comment:
We should move `index_header_lazy_loading_enabled` to `index_header` section at some point, if this experiment survives.

Reply:
To avoid introducing a breaking change, I would rather do the opposite and use **YAML inline** for the new index header config struct in the bucket stores config. This would allow us to have a dedicated struct for index header config in the code, but not having to move existing stable flags.

Reply:
Actually nevermind, I'll leave it as is for now and wait to see if our various experiments with index header changes survive.